### PR TITLE
Detect when popped to root on iOS after clicking on Tab

### LINKED ISSUE #9006: [Bug] Unable to open a new Page for the second time in Xamarin.Forms Shell Tabbar
### Description
In Xamarin.Forms Shell Tabbar, one cannot navigate to the first page that is opened in a Tab.
Supposing you are on Tab1 of the Tabbar and that Tab has 3 pages under the hierarchy and you click on a button which navigates to Tab1/Page1. Now you Tap on Tab1 Icon on the Tabbar and you are navigated back on Tab1. Now if you click on the same button to navigate to  Tab1/Page1  you will not be navigated. Now if you kill the App and launch again and you click on a button which navigates to Tab1/Page2, you get navigated to Page2 and you come back by clicking at the Tab1 on the Tabbar and not the back Icon on the Navigation Bar, in this case you cannot navigate to Tab1/Page2 again ever unless you kill the App.

### Steps to Reproduce

1. Launch a Xamarin Forms Shell App and by default you are on Tab1.
2. Click on the button that says "Click here to go to Page 1A" and you are navigated to Tab1/Page1A.
3. Click on Tab1 icon at the Tabbar, you will be navigated back to Tab1.
4. Click on the button that says "Click here to go to Page 1A" and you will not be able to navigate to Tab1/Page1A.

### Expected Behavior

1. Launch a Xamarin Forms Shell App and by default you are on Tab1.
2. Click on the button that says "Click here to go to Page 1A" and you are navigated to Tab1/Page1A.
3. Click on Tab1 icon at the Tabbar, you will be navigated back to Tab1.
4. Click on the button that says "Click here to go to Page 1A" and you **should** navigate to Tab1/Page1A.

### Basic Information

- Version with issue: Xamarin.Forms 4.4.0.991265
- Last known good version: NA
- IDE: Visual Studio 
- Platform Target Frameworks: 
  - iOS:  

- Affected Devices:
All

### Screenshots
ScreenRecording2019-12-23at41030

### Reproduction Link
Here is a link to the sample repo for you to verify.
https://github.com/nikhildpardasani/RoutingInShell

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###
Detect when PopToRoot occurs from clicking on the active tab and propagate these changes up to cross platform code

### Issues Resolved ### 

- fixes #9006 

### Platforms Affected ### 
- iOS

### Testing Procedure ###
- uitests includeed

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Core/Shell/IShellSectionController.cs [hunk 1]
```diff
@@ -21,8 +21,7 @@ public interface IShellSectionController : IElementController
 
 		void SendInsetChanged(Thickness inset, double tabThickness);
 
-		void SendPopped();
```
Comment (rmarinho) [inline Xamarin.Forms.Core/Shell/IShellSectionController.cs]: Isn't this a breaking change? 
We removed `SendPopped()` and `SendPopped(page)` and added new ones.

Comment (PureWeen) [reply Xamarin.Forms.Core/Shell/IShellSectionController.cs]: Controllers are meant to be broken right?

Comment (samhouts) [reply Xamarin.Forms.Core/Shell/IShellSectionController.cs]: It's definitely a breaking change if we expect users to subclass `ShellSection`.

Comment (PureWeen) [reply Xamarin.Forms.Core/Shell/IShellSectionController.cs]: Subclassing is fine and won't be breaking. 

It's only breaking for people that have used a custom class that's not ShellSection and added IShellSectionController to it. 

This is no different than any of the other controller interfaces we have on Button, Image, etc..

Comment (PureWeen) [reply Xamarin.Forms.Core/Shell/IShellSectionController.cs]: That being said :-p Added them back with some attributes

**Pull request merged**